engine/paima-providers/src/evm.ts

```diff
@@ -161,17 +161,19 @@ export class EvmProvider implements IProvider<EvmApi> {
       const walletChain = await this.conn.api.request({ method: 'eth_chainId' });
       return parseInt(walletChain as string, 16) === parseInt(this.gameInfo.gameChainId, 16);
     } catch (e: any) {
-      throw new ProviderApiError(`[verifyWalletChain] error: ${e?.message}`);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      throw new ProviderApiError(`[verifyWalletChain] error: ${e?.message}`, e?.code);
     }
   };
   addChain = async (newChain: AddEthereumChainParameter): Promise<void> => {
     try {
       await this.conn.api.request({
-        method: 'eth_chainId',
+        method: 'wallet_addEthereumChain',
```

Comment:
Oops, my bad. This was probably a copy-paste mistake

On a related note, this function if called by `async function switchChain()`, but I realize now that the function also never calls `switchChain` after it adds the network which maybe causes some issues in some wallets